File: sdkwork-ai-sdk-swift/Sources/API/ChatCompletionsApi.swift
import Foundation

public class ChatCompletionsApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Get chat completion
    public func getChatCompletion(completionId: String) async throws -> ChatCompletionResponse? {
        let response = try await client.get(ApiPaths.aiPath("/management/chat/completions/\(completionId)"))
        return response as? ChatCompletionResponse
    }

    /// Update chat completion
    public func updateChatCompletion(completionId: String, body: UpdateChatCompletionRequest) async throws -> ChatCompletionResponse? {
        let response = try await client.post(ApiPaths.aiPath("/management/chat/completions/\(completionId)"), body: body)
        return response as? ChatCompletionResponse
    }

    /// Delete chat completion
    public func deleteChatCompletion(completionId: String) async throws -> ChatCompletionDeleteResponse? {
        let response = try await client.delete(ApiPaths.aiPath("/management/chat/completions/\(completionId)"))
        return response as? ChatCompletionDeleteResponse
    }

    /// Get chat messages
    public func getChatMessages(completionId: String, params: [String: Any]? = nil) async throws -> ChatMessageList? {
        let response = try await client.get(ApiPaths.aiPath("/management/chat/completions/\(completionId)/messages"), params: params)
        return response as? ChatMessageList
    }

    /// List chat completions
    public func listChatCompletions(params: [String: Any]? = nil) async throws -> ChatCompletionList? {
        let response = try await client.get(ApiPaths.aiPath("/management/chat/completions"), params: params)
        return response as? ChatCompletionList
    }
}

File: sdkwork-ai-sdk-csharp/Api/ChatCompletionsApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class ChatCompletionsApi
    {
        private readonly HttpClient _client;

        public ChatCompletionsApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Get chat completion
        /// </summary>
        public async Task<ChatCompletionResponse?> GetChatCompletionAsync(string completionId)
        {
            return await _client.GetAsync<ChatCompletionResponse>(ApiPaths.AiPath($"/management/chat/completions/{completionId}"));
        }

        /// <summary>
        /// Update chat completion
        /// </summary>
        public async Task<ChatCompletionResponse?> UpdateChatCompletionAsync(string completionId, UpdateChatCompletionRequest body)
        {
            return await _client.PostAsync<ChatCompletionResponse>(ApiPaths.AiPath($"/management/chat/completions/{completionId}"), body);
        }

        /// <summary>
        /// Delete chat completion
        /// </summary>
        public async Task<ChatCompletionDeleteResponse?> DeleteChatCompletionAsync(string completionId)
        {
            return await _client.DeleteAsync<ChatCompletionDeleteResponse>(ApiPaths.AiPath($"/management/chat/completions/{completionId}"));
        }

        /// <summary>
        /// Get chat messages
        /// </summary>
        public async Task<ChatMessageList?> GetChatMessagesAsync(string completionId, Dictionary<string, object>? query = null)
        {
            return await _client.GetAsync<ChatMessageList>(ApiPaths.AiPath($"/management/chat/completions/{completionId}/messages"), query);
        }

        /// <summary>
        /// List chat completions
        /// </summary>
        public async Task<ChatCompletionList?> ListChatCompletionsAsync(Dictionary<string, object>? query = null)
        {
            return await _client.GetAsync<ChatCompletionList>(ApiPaths.AiPath("/management/chat/completions"), query);
        }
    }
}
